### ISSUE TITLE
HTTP File server for PFS

### ISSUE THREAD

**Issue by gabrielgrant** [pachyderm/pachyderm#2030]
In talking with @jdoliner and @sjezewski we decided to add an HTTP file server interface to pachd to be used initially by the UI.

Initially, this will allow downloading files from PFS by doing
```
GET http://pachd:658/pfs/files/<repo-name>/<commit-id>/<...path/to/file...>
```

A second stage will allow adding files to a commit by POSTing to the same URL. If a POST request is issued for a commit which is already closed, the servers should return a "405 Method Not Allowed" HTTP error

Open question: how should this interact with the new access controls being added in #2024?

Any REST API support beyond downloading files from known commits is outside the scope of this mini-project

**Comment by jdoliner** [comment]
I'd prefer the url be:

```
GET http://pachd:658/pfs/<repo-name>/<commit-id>/<...path/to/file...>
```

seems bit redundant to have `files` when the `f` in `pfs` already stands for file. Also, even though this isn't adding support for `repo`s or `commit`s I think we'll want those to be under `pfs/<repo-name>` and `pfs/<commit-id>`.

**Comment by gabrielgrant** [comment]
Ah, I thought you'd asked for it to be beneath `files` (or something similar) to allow other API endpoints to be added later.

Shouldn't commit ids be namespaced to repos? (ie `pfs/<repo-name>/<commit-id>`)

Though really, a "proper" RESTful endpoint structure usually follows a "<object-type-noun>/<object-identifer>" structure, meaning it should be something more like:
```
GET http://pachd:658/pfs/repos/<repo-name>/commits/<commit-id>/file/<...path/to/file...>
```

Where 
```
GET http://pachd:658/pfs/repos/
```
would return a list of repos and
```
GET http://pachd:658/pfs/repos/<repo-name>/commits/
```
returns a list of commits within a repo.

This lets you have other nouns without creating conflicts, like
```
GET http://pachd:658/pfs/repos/<repo-name>/tags/
```

**Comment by gabrielgrant** [comment]
This is implemented for files in #2043 
As discussed with @sjezewski we would like to also be able to download directoiries and entire repos as as a single .zip or .tar.gz file
